includes/emails/class-wc-email.php

```diff
@@ -250,14 +250,23 @@ public function handle_multipart( $mailer ) {
 	/**
 	 * Format email string.
 	 *
-	 * @param mixed $string
+	 * @param mixed $string Text to replace placeholders in.
 	 * @return string
 	 */
 	public function format_string( $string ) {
-		// Legacy placeholders. @todo deprecate in 4.0.0.
+		$find    = array_keys( $this->placeholders );
+		$replace = array_values( $this->placeholders );
+
+		// If using legacy find replace, add those to our find/replace arrays first. @todo deprecate in 4.0.0.
+		if ( isset( $this->find, $this->replace ) ) {
```

Comment:
`isset` on an empty array (which is initialized by default in this class) will return true. Do we want `! empty`?

Reply:
Would it hurt to remove the variables from the class?

Reply:
Probably not. I am thinking if we should just use `in_array` instead. Can legacy `find`/`replace` be strings?

Reply:
Arrays only. Core no longer sets them. If a plugin extending woocommerce used them, they would be set, so if we can remove them from the class, great.

Reply:
Then in my opinion we can leave the `isset`s and remove the `find`/`replace` definitions of this class 👍